include/swift/Runtime/Concurrency.h

```diff
@@ -107,6 +107,10 @@ void swift_task_dealloc(AsyncTask *task, void *ptr);
 SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
 void swift_task_cancel(AsyncTask *task);
 
+/// Get 'active' AsyncTask, depending on platform this may use thread local storage.
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+AsyncTask* swift_task_get_active();
+
```

Comment:
So this in practice replaces/does what `Builtin.getCurrentAsyncTask` did before, but works from sync context too.

Should we instead fix:

```swift
llvm::Value *IRGenFunction::getAsyncTask() {
  assert(isAsync()); // remove
  return Builder.CreateLoad(asyncTaskLocation); // replace with loading the ActiveTask TL
}
```

to do what the impl of this func does? I assume so, but my IRGen skills are a bit weak to do it 😢

Reply:
I think it's fine to keep this as a runtime entry point for now; we can move away from that later if we want.